src/partition/repart.c

```diff
@@ -1973,16 +1974,20 @@ static int context_load_partition_table(Context *context) {
         assert(context->end == UINT64_MAX);
         assert(context->total == UINT64_MAX);
 
+        c = fdisk_new_context();
+        if (!c)
+                return log_oom();
+
+        r = fdisk_save_user_sector_size(c, /* phy= */ 0, arg_sector_size);
+        if (r < 0)
+                return log_error_errno(r, "Failed to set sector size: %m");
```

Comment:
unless I am mistaken `r` is not a real error code here, but libfdisk just returns -1 on failure always... maybe pass `SYNTHETIC_ERRNO(EIO)` as first arg here, for lack of a better error code to return